HW1/P2/P2b.py

```diff
@@ -1,3 +1,30 @@
+# P2 solution part b
 from P2 import *
+import matplotlib.pyplot as plt
+from pyspark import SparkConf, SparkContext
 
-# Your code here
+def IJtoXY(I, J, pixel):
+    return J * 4.0 / pixel - 2, I * 4.0 / pixel - 2 
+
+if __name__ == '__main__':
+    
+    conf = SparkConf().setAppName('KaiSquare')
+    sc = SparkContext(conf = conf)
+
+    pixel = 2000
+    
+    I = sc.parallelize(range(pixel + 1))
+    J = sc.parallelize(range(pixel + 1))
+
+    IJ = I.cartesian(J)
+    IJ = IJ.partitionBy(100) # this is different from part a
```

Comment:
Wow, different approach. I just used random partition, and did not think about repartitioning after cartesian product. But I also wonder as "foo-bar-baz-qux" mentioned, there might be some chunks with similar values causing high iterations.